### PR TITLE
feature/v2-updates

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Fivetran created PR

**What change(s) does this PR introduce?** 

This PR includes the following updates:
- Addition of the following new models:
  - `pinterest_ads__advertiser_report`
  - `pinterest_ads__campaign_report`
  - `pinterest_ads__ad_group_report`
  - `pinterest_ads__keyword_report`
  - `pinterest_ads__pin_promotion_report`
  - `pinterest_ads__url_report`
- Added testing for each end model to ensure granularity and accuracy of the modeled data.
- README and yml documentation updates

**Did you update the CHANGELOG?** 

- [ ] Yes

**Does this PR introduce a breaking change?**

- [X] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [X] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [X] Yes, will also close out Issue #16 
- [X] No 

**How did you test the PR changes?** 

 

- [X] CircleCi  
- [X] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [X] Postgres
- [X] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🏊‍♂️ 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Review by fivetran-sheringuyen — COMMENTED** [review]
Hey @fivetran-joemarkiewicz, awesome PR! 

I left some in-line notes and had a couple of questions:
- In this package, I noticed in the `.yml` file that sometimes you do a DOCS reference for a definition and sometimes you have the definition explicitly in the `.yml`, is there a reason for that? E.g. some `campaign_id` fields will have the definition and some will refer to DOCS. 
- Can you talk a little as to why there's a non_null test for `date_day` across your models?
- Looks like similar to the source package, you can close PR #19 as well as delete the branch for spark :)

File: DECISIONLOG.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Decision Log
+## UTM Report Filtering
+This package contains a `pinterest_ads__url_report` which provides daily metrics for your utm compatible ads. It is important to note that not all Ads within Pinterest's `pin_promotion_report` source table do not leverage utm parameters. Therefore, this package takes an opinionated approach to filter out any records that do not contain utm parameters or leverage a url within the promoted pin.
+
+If you would like to leverage a report that contains all promoted pins and their daily metrics, I would suggest you leverage the `pinterest_ads__ad_report` which does not apply any filtering.
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply DECISIONLOG.md:5]: We > I. Updated!

File: models/pinterest.yml [hunk 2]
```diff
@@ -1,16 +1,41 @@
 version: 2
 
 models:
-  - name: pinterest_ads__ad_adapter
+  - name: pinterest_ads__url_report
     description: Each record in this table represents the daily performance of ads at the ad group and url level.
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - pin_promotion_id
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/pinterest.yml:9]: Made the same updates from the source package here to account for the correct grains.

File: models/pinterest.yml [hunk 3]
```diff
@@ -35,47 +60,200 @@ models:
...
         description: The amount of spend that occurred on the day of the record.
 
-  - name: pinterest_ads__ad_group_ad_report
+  - name: pinterest_ads__ad_group_report
     description: Each record in this table represents the daily performance of ads at the ad group level.
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - ad_group_id
+            - date_day
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/pinterest.yml:76]: Updated. Per our and Pinterest's docs the ad group is unique across campaigns but can be reused across advertisers. Updated the grain test to reflect this.

File: models/pinterest.yml [hunk 4]
```diff
@@ -35,47 +60,200 @@ models:
...
+      - name: impressions
+        description: "{{ doc('impressions') }}"
+
+  - name: pinterest_ads__keyword_report
+    description: Each record in this table represents the daily performance of a keyword.
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - keyword_id
+            - ad_group_id
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/pinterest.yml:173]: Updated. Per our and Pinterests docs, the keyword can be reused across the ad group, campaign, and advertiser. Updated the grain test to reflect this.

File: dbt_project.yml [hunk 5]
```diff
@@ -8,10 +8,22 @@ models:
     +materialized: table
     intermediate:
       +materialized: ephemeral
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply dbt_project.yml:10]: Very true! removed.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks for the review @fivetran-sheringuyen! I was able to address your inline comments and see below for my responses to your broader questions:

> In this package, I noticed in the .yml file that sometimes you do a DOCS reference for a definition and sometimes you have the definition explicitly in the .yml, is there a reason for that? E.g. some campaign_id fields will have the definition and some will refer to DOCS.

- This was an oversight on my part. I didn't update the previous docs to use the docs.md I created from the source. I just updated the previous descriptions to fit with the standard across models of using the docs.md when available.

> Can you talk a little as to why there's a non_null test for date_day across your models?

- Following our team discussions. I added the not null tests to the day and lowest grain fields to ensure data completeness.

> Looks like similar to the source package, you can close PR https://github.com/fivetran/dbt_pinterest/pull/19 as well as delete the branch for spark :)

- For sure! I closed the PR and deleted the stale branch.

**Review by fivetran-sheringuyen — APPROVED** [review]
Awesome work, thanks for the updates @fivetran-joemarkiewicz !

**Pull request merged**